plugins/manipulator.cpp

```diff
@@ -867,7 +1797,8 @@ class viewscreen_unitbatchopst : public dfhack_viewscreen {
             else if (events->count(interface_key::SELECT))
             {
                 apply((cur_page == NICKNAME) ? unit_ops::set_nickname : unit_ops::set_profname, entry, &formatter);
-                select_page(MENU);
+                Screen::dismiss(this);
```

Comment:
This is somewhat of a personal preference. The original behavior here was intended to allow multiple changes to be made without having to re-enter the menu every time. Of course, the menu doesn't have many options now, so maybe it's not useful behavior. Just seems unrelated.

Reply:
Few users will want to change both nickname and professions consequtively, and those who might do wont be repeating that operation often so the escape keypress involved matters little for them. But if you to go through the list nicking units one by one which is a natural action, its much nicer to not have an escape keypress in the way.